### PR TITLE
Feature/dcc 5555 validation

### PR DESCRIPTION
**Pull request by ayang99**

### CODE REVIEW

File: sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/config/AsyncProcessingConfig.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@EnableAsync
+public class AsyncProcessingConfig extends AsyncConfigurerSupport {
+	
+	@Value("${validation.threads.core}")
+  private int corePoolSize;
+
+	@Value("${validation.threads.max}")
+  private int maxPoolSize;
+
+	@Value("${validation.queue.capacity}")
```
Comment (andricDu) [inline sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/config/AsyncProcessingConfig.java]: dem tabs!

File: sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/validation/SchemaValidator.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import lombok.val;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Potentially extract a Validator interface if we want to pursue a Strategy pattern of multiple
+ * validation rules or something
+ *
+ */
+@Slf4j
+public class SchemaValidator {
```
Comment (andricDu) [inline sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/validation/SchemaValidator.java]: this class is using tabs

File: sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/service/StatusService.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  	return status;
+  }
+  
+  public void updateAsValid(String studyId, String uploadId) {
+  	statusRepository.update(uploadId, studyId, SubmissionStatus.VALIDATED, "");
+  }
+  
+  public void updateAsInvalid(String studyId, String uploadId, String errorMessages) {
+  	statusRepository.update(uploadId, studyId, SubmissionStatus.VALIDATION_ERROR, errorMessages);
+  }
```
Comment (andricDu) [inline sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/service/StatusService.java]: \n

File: sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/service/RegistrationService.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+
+import org.skife.jdbi.v2.exceptions.UnableToExecuteStatementException;
+
+@Slf4j
+@Service
+public class RegistrationService {
+
+	@Autowired
+	private SchemaValidator validator;
```
Comment (andricDu) [inline sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/service/RegistrationService.java]: tabs are here too

File: sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/validation/ValidationResponse.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+		this.messages = messages;
+		if ((messages != null) && !messages.isEmpty()) {
+			valid = false;
+		}
+	}
+	
+	public String getValidationErrors() {
+		val errorMessages = messages.stream()
+				.map(ValidationMessage::getMessage)
+				.collect(Collectors.joining("|"));
```
Comment (andricDu) [inline sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/validation/ValidationResponse.java]: static import for `Collectors.joining`

File: sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/repository/mapper/StatusMapper.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.LocalDateTime;
+
+public class StatusMapper implements ResultSetMapper<SubmissionStatus> {
+
+  public SubmissionStatus map(int index, ResultSet rs, StatementContext ctx) throws SQLException
+  { // I prefer braces on next line when declaring exception throws in method signature - Dušan
+  	SubmissionStatus status = new SubmissionStatus();
+  	status.withUploadId(rs.getString("id"))
```
Comment (andricDu) [inline sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/repository/mapper/StatusMapper.java]: Any reason we are not using a builder for SubmissionStatus?

**Comment by ayang99** [comment]
I use the following two commands for testing (in Insomnia)

curl --request POST \
  --url http://localhost:8080/studies/ABC123/analyses/sequencingread/fe5346ba-24de-48c7-9824-82dcea8a8a23 \
  --header ': ' \
  --header 'content-type: application/json' \
  --data '{
  "sequencingRead": {
    "aligned": true,
    "alignmentTool": "deserunt velit mollit",
    "insertSize": 28042512,
    "libraryStrategy": "RNA-Seq",
    "pairedEnd": false,
    "referenceGenome": "qui amet"
  },
  "study": {
    "name": "Duis sint",
    "organization": "enim",
    "description": "velit anim est",
    "donor": {
      "donorSubmitterId": "incididunt fugiat",
      "donorGender": "male",
      "specimen": {
        "specimenSubmitterId": "submissius plodicus",
        "specimenClass": "Normal",
        "specimenType": "Normal - blood derived",
        "sample": {
          "sampleSubmitterId": "est sed id",
          "sampleType": "RNA",
          "files": [
            {
              "fileName": "magna",
              "fileSize": 59857573,
              "fileType": "IDX",
              "fileMd5": "occaecat"
            },
            {
              "fileName": "deserunt fugiat enim ci",
              "fileSize": 8897683,
              "fileType": "FASTA",
              "fileMd5": "minim reprehenderit"
            },
            {
              "fileName": "incidid",
              "fileSize": 86761872,
              "fileType": "XML",
              "fileMd5": "ad exercitation"
            },
            {
              "fileName": "id",
              "fileSize": 21090,
              "fileType": "XML",
              "fileMd5": "in aliqua ad id"
            }
          ]
        }
      }
    }
  }
}'

and then:

curl --request GET \
  --url http://localhost:8080/studies/ABC123/statuses/fe5346ba-24de-48c7-9824-82dcea8a8a23 \
  --header ': '

The in-process h2 console is also enabled to allow inspection of the database tables: jdbc:h2:file:/tmp/db (the file name uses the ${java.io.tmpdir} property)

The sodalite-server is run with: -Djava.io.tmpdir=/tmp --security.basic.enabled=false

File: sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/repository/mapper/StatusMapper.java (hunk 6 above)
Comment (ayang99) [inline sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/repository/mapper/StatusMapper.java]: I followed the convention that the other POJO classes used (they were generated by the JSON Schema --> POJO converter.

They should all be revised the same way.

**Comment by andricDu** [comment]
I'll use this PR to fix the jenkins stuff.

**Comment by andricDu** [comment]
test this please

**Comment by andricDu** [comment]
add to whitelist

**Comment by andricDu** [comment]
retest this please

**Pull request merged**